ihmc-perception/src/main/java/us/ihmc/perception/gpuHeightMap/RapidHeightMapExtractorCUDA.java

```diff
@@ -0,0 +1,529 @@
+      }
+   }
+
+   public void initialize() throws URISyntaxException
+   {
+      stream = new CUstream_st();
+      cudart.cudaStreamCreate(stream);
+
+      // Load CUDA kernels
+      URL kernelPath = getClass().getResource("RapidHeightMapExtractor.cu");
+      heightMapCUDAProgram = new CUDAProgram(kernelPath);
+
+      String updateKernelName = "heightMapUpdateKernel";
+      String registerKernelName = "heightMapRegistrationKernel";
+      String croppingKernelName = "croppingKernel";
+      String preprocessKernelName = "preprocessImageKernel";
+
+      updateKernel = heightMapCUDAProgram.loadKernel(updateKernelName);
+      registerKernel = heightMapCUDAProgram.loadKernel(registerKernelName);
+      croppingKernel = heightMapCUDAProgram.loadKernel(croppingKernelName);
+      preprocessKernel = heightMapCUDAProgram.loadKernel(preprocessKernelName);
+
+      recomputeDerivedParameters();
+
+      cropWindowRectangle = new Rect((globalCellsPerAxis - heightMapParameters.getCropWindowSize()) / 2,
+                                     (globalCellsPerAxis - heightMapParameters.getCropWindowSize()) / 2,
+                                     heightMapParameters.getCropWindowSize(),
+                                     heightMapParameters.getCropWindowSize());
+
+      terrainMapData = new TerrainMapData(heightMapParameters.getCropWindowSize(), heightMapParameters.getCropWindowSize());
+
+      // Initialize matrices and images
+      denoisedHeightMapImage = new Mat(heightMapParameters.getCropWindowSize(), heightMapParameters.getCropWindowSize(), opencv_core.CV_16UC1);
+      steppableRegionAssignmentMat = new Mat(heightMapParameters.getCropWindowSize(), heightMapParameters.getCropWindowSize(), opencv_core.CV_16UC1);
+      steppableRegionRingMat = new Mat(heightMapParameters.getCropWindowSize(), heightMapParameters.getCropWindowSize(), opencv_core.CV_8UC1);
+
+      localHeightMapImage = new GpuMat(localCellsPerAxis, localCellsPerAxis, opencv_core.CV_16UC1);
+
+      transformedInputDepthImage = new GpuMat(inputDepthImage.cols(), inputDepthImage.rows(), opencv_core.CV_16UC1);
+
+      globalHeightMapImage = new GpuMat(globalCellsPerAxis, globalCellsPerAxis, opencv_core.CV_16UC1);
+
+      globalHeightVarianceImage = new GpuMat(globalCellsPerAxis, globalCellsPerAxis, opencv_core.CV_8UC1);
+
+      terrainCostImage = new GpuMat(globalCellsPerAxis, globalCellsPerAxis, opencv_core.CV_8UC1);
+
+      contactMapImage = new GpuMat(globalCellsPerAxis, globalCellsPerAxis, opencv_core.CV_8UC1);
+
+      sensorCroppedHeightMapImage = new GpuMat(heightMapParameters.getCropWindowSize(), heightMapParameters.getCropWindowSize(), opencv_core.CV_16UC1);
+
+      sensorCroppedTerrainCostImage = new GpuMat(heightMapParameters.getCropWindowSize(), heightMapParameters.getCropWindowSize(), opencv_core.CV_8UC1);
+
+      sensorCroppedContactMapImage = new GpuMat(heightMapParameters.getCropWindowSize(), heightMapParameters.getCropWindowSize(), opencv_core.CV_8UC1);
```

Comment:
I think it would be better if we didn't have extra new lines between each of these